fbw-a32nx/src/systems/failures/src/AltitudeFailureGenerator.ts

```diff
@@ -0,0 +1,95 @@
+import { NXDataStore } from '@flybywiresim/fbw-sdk';
+import { ArmingIndex, FailurePhases, FailuresAtOnceIndex, MaxFailuresIndex, RandomFailureGen } from './RandomFailureGen';
+import { FailuresOrchestrator } from './failures-orchestrator';
+
+export class FailureGeneratorAltitude {
+    private static settingName = 'EFB_FAILURE_GENERATOR_SETTING_ALTITUDE';
+
+    private static numberOfSettingsPerGenerator = 6;
+
+    private static uniqueGenPrefix = 'A';
+
+    private static failureGeneratorArmed :boolean[] = [];
+
+    private static doNotRepeatUntilTakeOff: boolean[] = [];
+
+    private static rolledDice:number[] = [];
+
+    private static didOnce : boolean = false;
+
+    private static altitudeConditionIndex = 3;
+
+    private static altitudeMinIndex = 4;
+
+    private static altitudeMaxIndex = 5;
+
+    private static resetMargin = 100;
+
+    static updateFailure(failureOrchestrator : FailuresOrchestrator) : void {
+        const failureGeneratorSetting = NXDataStore.get(FailureGeneratorAltitude.settingName, '');
+
+        if (!FailureGeneratorAltitude.didOnce) {
+            const generatorNumber = Math.floor(failureGeneratorSetting.split(',').length / FailureGeneratorAltitude.numberOfSettingsPerGenerator);
+            for (let i = 0; i < generatorNumber; i++) {
+                FailureGeneratorAltitude.failureGeneratorArmed[i] = false;
+                FailureGeneratorAltitude.doNotRepeatUntilTakeOff[i] = false;
+            }
+            FailureGeneratorAltitude.didOnce = true;
+        }
+
+        const settings : number[] = failureGeneratorSetting.split(',').map(((it) => parseFloat(it)));
```

Comment:
Yes, these settings are persistent. I keep the bare minimum in them. I want players to be able to configure something once and for all as sometimes it may take a bit of time to set it up